spotinst/oceancd_verification_template_args/fields_spotinst_oceancd_verification_template_args.go

```diff
@@ -0,0 +1,218 @@
+	"github.com/spotinst/spotinst-sdk-go/spotinst"
+	"github.com/spotinst/terraform-provider-spotinst/spotinst/commons"
+)
+
+func Setup(fieldsMap map[commons.FieldName]*commons.GenericField) {
+	fieldsMap[Args] = commons.NewGenericField(
+		commons.OceanCDVerificationTemplateArgs,
+		Args,
+		&schema.Schema{
+			Type:     schema.TypeSet,
+			Optional: true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					string(ArgName): {
+						Type:     schema.TypeString,
+						Required: true,
+					},
+
+					string(Value): {
+						Type:     schema.TypeString,
+						Optional: true,
+					},
+
+					string(ValueFrom): {
+						Type:     schema.TypeList,
+						Optional: true,
+						MaxItems: 1,
+						Elem: &schema.Resource{
+							Schema: map[string]*schema.Schema{
+								string(SecretKeyRef): {
+									Type:     schema.TypeList,
+									Optional: true,
+									MaxItems: 1,
+									Elem: &schema.Resource{
+										Schema: map[string]*schema.Schema{
+											string(Key): {
+												Type:     schema.TypeString,
+												Required: true,
+											},
+
+											string(KeyName): {
+												Type:     schema.TypeString,
+												Required: true,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			verificationTemplateWrapper := resourceObject.(*commons.OceanCDVerificationTemplateWrapper)
+			verificationTemplate := verificationTemplateWrapper.GetVerificationTemplate()
+
+			var argsResults []interface{} = nil
+			if verificationTemplate != nil && verificationTemplate.Args != nil {
+				args := verificationTemplate.Args
+				argsResults = flattenArgs(args)
+			}
+
+			if err := resourceData.Set(string(Args), argsResults); err != nil {
+				return fmt.Errorf(string(commons.FailureFieldReadPattern), string(Args), err)
+			}
+
+			return nil
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			verificationTemplateWrapper := resourceObject.(*commons.OceanCDVerificationTemplateWrapper)
+			verificationTemplate := verificationTemplateWrapper.GetVerificationTemplate()
+			if value, ok := resourceData.GetOkExists(string(Args)); ok {
+				if args, err := expandArgs(value); err != nil {
+					return err
+				} else {
+					verificationTemplate.SetArgs(args)
+				}
+			}
+			return nil
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			verificationTemplateWrapper := resourceObject.(*commons.OceanCDVerificationTemplateWrapper)
+			verificationTemplate := verificationTemplateWrapper.GetVerificationTemplate()
+			var result []*oceancd.Args = nil
+			if value, ok := resourceData.GetOkExists(string(Args)); ok {
+				if args, err := expandArgs(value); err != nil {
+					return err
+				} else {
+					result = args
+				}
+			}
+
+			if len(result) == 0 {
```

Comment:
logic to be re-checked

Reply:
Changed the Logic, This logic was also working but was not according to the latest code format